src/cargo/core/compiler/custom_build.rs

```diff
@@ -61,7 +61,7 @@ const OLD_CARGO_WARNING_SYNTAX: &str = "cargo:warning=";
 /// [the doc]: https://doc.rust-lang.org/nightly/cargo/reference/build-scripts.html#cargo-warning
 const NEW_CARGO_WARNING_SYNTAX: &str = "cargo::warning=";
 /// Contains the parsed output of a custom build script.
-#[derive(Clone, Debug, Hash, Default)]
+#[derive(Clone, Debug, Hash, Default, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
Also realized that there is an implication that these newly added fields would affect the order of any sorted units list. I don't think that would cause real issues though.

Reply:
True... if this is a significant concern we could move the new fields to the end.

Ultimately I don't think I have the context to decide if it is. I think it is almost certainly publicly exposed given comments like [this one](https://github.com/rust-lang/cargo//blob/master/src/cargo/core/compiler/unit_dependencies.rs#L140)), and where these fields currently are in the struct is probably high enough to change the order in the cases [described in this comment](https://github.com/rust-lang/cargo//blob/master/src/cargo/core/compiler/unit.rs#L98). I wouldn't have thought that the ordering of units was something that was guaranteed to be stable between cargo versions.

I'll hold off on updating the rustflags `Arc` to `Rc` until we've decided if we need to move it to the bottom since that might as well be one PR.